Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/transform/CoreX12AuditDataTransform.java

```diff
@@ -0,0 +1,467 @@
+            AuditDataTransformConstants.EVENT_OUTCOME_INDICATOR_SUCCESS, eventID);
+        return oEventIdentificationType;
+    }
+
+    public static AuditMessageType.ActiveParticipant getActiveParticipantSource(boolean isRequesting) {
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+        participant.setUserID(CoreX12AuditDataTransformConstants.ACTIVE_PARTICPANT_USER_ID_SOURCE);
+        participant.setAlternativeUserID(ManagementFactory.getRuntimeMXBean().getName());
+        try {
+            String ipAddr = InetAddress.getLocalHost().getHostAddress();
+            participant.setNetworkAccessPointID(ipAddr);
+            participant.setNetworkAccessPointTypeCode(AuditDataTransformConstants.NETWORK_ACCESSOR_PT_TYPE_CODE_IP);
+        } catch (UnknownHostException ex) {
+            LOG.warn("UnknownHostException thrown getting local host address.", ex);
+            return null;
+        }
+
+        participant.getRoleIDCode().add(AuditDataTransformHelper.createCodeValueType(CoreX12AuditDataTransformConstants.ACTIVE_PARTICPANT_ROLE_CODE_CDE, null,
+            AuditDataTransformConstants.EVENT_ID_CODE_SYS_NAME_DOC, CoreX12AuditDataTransformConstants.ACTIVE_PARTICPANT_ROLE_CODE_SOURCE_DISPLAY_NAME));
+        participant.setUserIsRequestor(isRequesting ? Boolean.TRUE : Boolean.FALSE);
+        return participant;
+    }
+
+    protected static AuditMessageType.ActiveParticipant getActiveParticipantDestination(NhinTargetSystemType target) {
+        String strUrl = null;
+        String strHost = null;
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+        if (null != target) {
+            if (NullChecker.isNotNullish(target.getUrl())) {
+                strUrl = target.getUrl();
+            } else {
+                ConnectionManagerCache cm = ConnectionManagerCache.getInstance();
+                try {
+                    strUrl = cm.getEndpointURLFromNhinTarget(target, NhincConstants.NHIN_CORE_X12DS_REALTIME_SECURED_SERVICE_NAME);
+                } catch (ConnectionManagerException ex) {
+                    LOG.error(ex);
+                }
+            }
+        }
+
+        if (null != strUrl) {
+            try {
+                URL url = new URL(strUrl);
+                participant.setUserID(strUrl);
+                strHost = url.getHost();
+                participant.setNetworkAccessPointID(strHost);
+                if (Character.isDigit(strHost.charAt(0))) {
+                    participant.setNetworkAccessPointTypeCode(AuditDataTransformConstants.NETWORK_ACCESSOR_PT_TYPE_CODE_IP);
+                } else {
+                    participant.setNetworkAccessPointTypeCode(CoreX12AuditDataTransformConstants.NETWORK_ACCESSOR_PT_TYPE_CODE_NAME);
+                }
+            } catch (MalformedURLException ex) {
+                LOG.error(ex);
```

Comment:
We don't really handle this exception here.  I would take what's in the below else and pull out a private method and then call that from this catch if an exception happens.  that way the participant is set on error.